packages/starknet-snap/src/executeTxn.ts

```diff
@@ -27,7 +33,15 @@ export async function executeTxn(params: ApiParams) {
       addressIndex,
     } = await getKeysFromAddress(keyDeriver, network, state, senderAddress);
 
-    if (await isUpgradeRequired(network, senderAddress)) {
+    const { upgradeRequired, deployRequired } = await getCorrectContractAddress(network, publicKey, state);
+
+    if (upgradeRequired && deployRequired) {
+      // Edge case force cairo0 deploy because non-zero balance
+      await showDeployRequestModal(wallet);
+      throw new Error('Deploy required');
```

Comment:
nic:
can we throw the Error like:

throw new Error('Cairo zero contract address `xxxx`'s balance is not empty, deploy required');